### PR TITLE
MRG: Omit 2.6, fix image_tester deprecation

### LINKED ISSUE #1020: Traceback: wxGLCanvasBase::SetCurrent(): can't make hidden GL canvas current
This is with wxpython 3.0.2.0 and several of the examples that I tried.

C:...\vispy\examples\basics\plotting>python scatter_histogram.py
INFO: Could not import backend "PyQt4":
No module named PyQt4
INFO: Could not import backend "PyQt5":
No module named PyQt5
INFO: Could not import backend "PySide":
No module named PySide
INFO: Could not import backend "Pyglet":
No module named pyglet
INFO: Could not import backend "Glfw":
GLFW library not found
INFO: Could not import backend "SDL2":
No module named sdl2
Traceback (most recent call last):
  File "scatter_histogram.py", line 18, in <module>
    fig = vp.Fig(show=False)
  File "C:\Python27\lib\site-packages\vispy-0.5.0.dev0-py2.7.egg\vispy\plot\fig.py", line 39, in **init**
    show=show, size=size)
  File "C:\Python27\lib\site-packages\vispy-0.5.0.dev0-py2.7.egg\vispy\scene\canvas.py", line 128, in **init**
    always_on_top, px_scale)
  File "C:\Python27\lib\site-packages\vispy-0.5.0.dev0-py2.7.egg\vispy\app\canvas.py", line 206, in **init**
    self.set_current()
  File "C:\Python27\lib\site-packages\vispy-0.5.0.dev0-py2.7.egg\vispy\app\canvas.py", line 406, in set_current
    self._backend._vispy_set_current()
  File "C:\Python27\lib\site-packages\vispy-0.5.0.dev0-py2.7.egg\vispy\app\backends_wx.py", line 294, in _vispy_set_current
    self.SetCurrent(self._gl_context)
  File "C:\Python27\lib\site-packages\wx-3.0-msw\wx\glcanvas.py", line 140, in SetCurrent
    return _glcanvas.GLCanvas_SetCurrent(*args)
wx._core.PyAssertionError: C++ assertion "IsShown()" failed at ....\src\common\glcmn.cpp(59) in wxGLCanvasBase::SetCurrent(): can't make hidden GL canvas current

This is with 64-bit Python 2.7.10 on windows 7.  Installed packages:

C:...>pip list
backports.ssl-match-hostname (3.4.0.2)
certifi (2015.4.28)
decorator (3.4.2)
functools32 (3.2.3.post2)
ipython (3.2.1)
Jinja2 (2.7.3)
jsonschema (2.5.1)
MarkupSafe (0.23)
matplotlib (1.4.3)
numpy (1.9.2)
pip (7.1.0)
py2exe (0.6.10a1)
PyAudio (0.2.8)
PyOpenGL (3.1.1a1)
PyOpenGL-accelerate (3.1.1a1)
pyparsing (2.0.3)
python-dateutil (2.4.2)
pytz (2015.4)
pywin32 (219)
pyzmq (14.7.0)
scipy (0.16.0rc1)
setuptools (16.0)
six (1.9.0)
tornado (4.2.1)
vispy (0.5.0.dev0)
wxPython (3.0.2.0)
wxPython-common (3.0.2.0)

### PR DESCRIPTION
**Pull request by larsoner**
I saw [this log](https://travis-ci.org/vispy/vispy/jobs/272409672) and thought it could be cleaned up a bit. Hopefully this does it.

Closes #1020.

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@larsoner Somehow the bundled `six` doesn't get picked up and I can't immediately see why. 😢

**Comment by larsoner** [comment]
Lots of CI fixes in this one. Ready for review/merge from my end. No need to look at `vispy/ext/_bundled/*` file changes because they are just copy-paste updates.

**Comment by larsoner** [comment]
(This drops 2.6 support and adds 3.6 support BTW.)

File: vispy/visuals/tests/test_arrows.py [hunk 1]
```diff
@@ -58,18 +56,11 @@ def test_arrow_draw():
...
-    old_numpy = LooseVersion(np.__version__) < '1.8'
-    if os.getenv('TRAVIS', 'false') == 'true' and (sys.version[:3] == '2.6' or
-                                                   old_numpy):
-        # TODO: Fix this (issue #1042
-        raise SkipTest('Travis fails due to FB stack problem')
-
     with TestingCanvas() as c:
+        if os.getenv('TRAVIS', 'false') == 'true' and \
+                c.app.backend_name == 'pyqt4':
+            # TODO: Fix this (issue #1042
```
Comment (kmuehlbauer) [inline vispy/visuals/tests/test_arrows.py]: ### #1042 mentioned python26. Is this still an issue?

**Comment by kmuehlbauer** [comment]
@larsoner First of all, great work!

I tried to follow all your of changes. I didn't catch any obvious while going through the changed files.

What I found so far in the travis runs:

- PYTHON=2.7 DEPS=full TEST=osmesa NUMPY="1.9"
https://travis-ci.org/vispy/vispy/jobs/273030781#L2817
    - https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/1410960
    - might be related with the provided osmesa-libs

- deprecation warning:
https://travis-ci.org/vispy/vispy/jobs/273030780#L3151

- flake8 not tested
https://travis-ci.org/vispy/vispy/jobs/273030774#L2447

I didn't look into appveyor yet.

**Comment by kmuehlbauer** [comment]
@larsoner Do you know what is needed to get backends (eg. qt5) tested in python 3.6? Or what holds us from doing so?

**Comment by larsoner** [comment]
Qt5 wouldn't even create a context on Linux, so it will probably be a pain -- see #960

That OSMesa I compiled a long time ago so it might need some updating, not sure.

The deprecation stuff we should deal with separately, it will require refactoring our testing code. I'll try to fix the Flake stuff, though, that should run.

**Comment by astrofrog** [comment]
Qt5 from the defaults channel doesn't yet support OpenGL: https://github.com/ContinuumIO/anaconda-issues/issues/1267

But the version from conda-forge does, so could use that channel?

**Comment by larsoner** [comment]
Perhaps, but let's do it in another PR :)

**Comment by larsoner** [comment]
Flake8 should be fixed. I don't think this fixes everything but it at least improves the situation and gets more testing done. @astrofrog @kmuehlbauer okay to merge once the CIs are happy?

**Comment by kmuehlbauer** [comment]
@larsoner Yes, sure go ahead. But we should not forget the other issues (eg. osmesa). BTW @astrofrog @davidh-ssec do you know, if we can utilise osmesa from conda-forge (mesalib)?

**Pull request merged**